lib/jxl/enc_fast_lossless.cc

```diff
@@ -3425,7 +3425,7 @@ void PrepareDCGlobalPalette(bool is_single_group, size_t width, size_t height,
   row_encoder.t = &encoder;
   encoder.output = output;
   encoder.code = &code[0];
-  int16_t p[4][32 + 1024] = {};
+  int16_t p[4][32 + 2048] = {};
```

Comment:
Is this needed? I don't remember exactly how this code works but I assume the size of this array should be something like `kMaxColors + kChunkSize` (and possibly the `+ kChunkSize` was only needed there because of potential padding, which is now no longer needed).

Reply:
Ah, you are right, 1024 would have been enough - I think we should instead static_assert that kMaxColors < 1024 then.

Reply:
Any reason why we can't just change it to this?
```suggestion
  int16_t p[4][kMaxColors] = {};
```

That way we don't need any static_asserts, and we don't allocate an array that is bigger than needed (`kMaxColors` is 512...

Reply:
This, certainly not. kMaxColors + 32 should work fine though :)

Reply:
Where does the 32 come from?

Reply:
kChunkSize * 2, but this might actually not be 32... Then again we also have a +16 below, which should be +kChunkSize instead.

Reply:
ok, I'm changing back to 1024 as suggested below. The change of the size of `p` and the `+16` below is somewhat unrelated to the current fix, so perhaps lets do this in another pr?!

Reply:
Sounds good!

Reply:
Can you fix this in another pr, @jonsneyers?

Reply:
sure